packages/dds/tree/src/feature-libraries/modular-schema/discrepancies.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * @public
+ */
+export interface AllowedTypeIncompatibility {
+	identifier: string;
+	mismatch: "allowedTypes";
+	/**
+	 * List of allowed type identifiers in viewed schema
+	 */
+	view: string[];
+	/**
+	 * List of allowed type identifiers in stored schema
+	 */
+	stored: string[];
+}
+
+/**
+ * @public
+ */
+export type SchemaFactoryFieldKind = "required" | "optional" | "array" | "identifier";
+
+/**
+ * @public
+ */
+export interface FieldKindIncompatibility {
+	identifier: string;
+	mismatch: "fieldKind";
+	view: SchemaFactoryFieldKind | undefined;
+	stored: SchemaFactoryFieldKind | undefined;
+}
+
+/**
+ * @public
+ */
+export type FieldIncompatibility = AllowedTypeIncompatibility | FieldKindIncompatibility;
+
+/**
+ * @public
+ */
+export type SchemaFactoryNodeKind = "object" | "array" | "map";
+
+/**
+ * @public
+ */
+export interface NodeKindIncompatibility {
```

Comment:
none of these types are package exported so they shouldn't be marked public.

they're going to cross module boundary, so we should have some doc comments explaining semantics.